transformer/classes/dockerfile/pythondockerfilegenerator.go

```diff
@@ -0,0 +1,248 @@
+	return t.Config, t.Env
+}
+
+// findMainScripts returns the path of .py files having the main function
+func findMainScripts(pythonFilesPath []string) ([]string, error) {
+	if len(pythonFilesPath) == 0 {
+		return []string{}, nil
+	}
+	var pythonMainFiles []string
+	for _, pythonFilePath := range pythonFilesPath {
+		pythonFile, err := os.Open(pythonFilePath)
+		if err != nil {
+			logrus.Debugf("Failed to open the file %s", pythonFilePath)
+		}
+		scanner := bufio.NewScanner(pythonFile)
+		scanner.Split(bufio.ScanLines)
+		for scanner.Scan() {
+			if pythonMainRegex.MatchString(scanner.Text()) {
+				pythonMainFiles = append(pythonMainFiles, pythonFilePath)
+			}
+		}
+	}
+	if len(pythonMainFiles) == 0 {
+		return []string{}, fmt.Errorf("could not find the main function in python files")
+	}
+	return pythonMainFiles, nil
+}
+
+// findDjangoDependency checks for django dependency in the requirements.txt file
+func findDjangoDependency(reqTxtFilePath string) bool {
+	reqTxtFile, err := os.Open(reqTxtFilePath)
+	if err != nil {
+		logrus.Debugf("Failed to open the file %s", reqTxtFilePath)
+	}
+	scanner := bufio.NewScanner(reqTxtFile)
+	scanner.Split(bufio.ScanLines)
+	for scanner.Scan() {
+		if djangoDependencyRegex.MatchString(scanner.Text()) {
+			return true
+		}
+	}
+	return false
+}
+
+// getMainPythonFileForService returns the main file used by a service
+func getMainPythonFileForService(mainPythonFilesPath []string, baseDir string, serviceName string) string {
+	var mainPythonFilesRelPath []string
+	for _, mainPythonFilePath := range mainPythonFilesPath {
+		if mainPythonFileRelPath, err := filepath.Rel(baseDir, mainPythonFilePath); err == nil {
+			mainPythonFilesRelPath = append(mainPythonFilesRelPath, mainPythonFileRelPath)
+		}
+	}
+	return qaengine.FetchSelectAnswer(common.ConfigServicesKey+common.Delim+serviceName+common.Delim+common.ConfigMainPythonFileForServiceKeySegment, fmt.Sprintf("Select the main file to be used for the service %s :", serviceName), []string{fmt.Sprintf("Selected main file will be used for the service %s", serviceName)}, mainPythonFilesRelPath[0], mainPythonFilesRelPath)
+}
+
+// DirectoryDetect runs detect in each sub directory
+func (t *PythonDockerfileGenerator) DirectoryDetect(dir string) (services map[string][]transformertypes.Artifact, err error) {
+
+	pythonFiles, err := common.GetFilesByExtInCurrDir(dir, []string{pythonExt})
+	if err != nil {
+		logrus.Errorf("Error while finding python files %s", err)
+		return nil, nil
+	}
+	if len(pythonFiles) == 0 {
+		return nil, nil
+	}
+	mainPythonFilesPath, err := findMainScripts(pythonFiles)
+	if err != nil {
+		return nil, err
+	}
+	if len(mainPythonFilesPath) != 0 {
+		var serviceName, requirementsTxtPath string
+		var isDjangoProject bool
+		var detectedPorts []int32
+		detectedPorts = append(detectedPorts, 8080) //TODO: Write parser to parse and identify port
+		requirementsTxtFiles, err := common.GetFilesByName(dir, []string{requirementsTxtFile}, nil)
```

Comment:
Shouldn't we be using GetFilesInCurrentDirectory? Generally requirements.txt is placed only in the base directory of the project. isn't it?

Reply:
Sure, I have changed it to the GetFilesInCurrentDirectory to look for the file only in the base directory.